src/build/partition.jl

```diff
@@ -173,12 +207,52 @@ function _partition_duty0!(d::Dict, set::Dict)
 end
 
 
+"""
+`fd(yr,g,fd)`, final demand, and
+`fs(yr,g)`, household supply
+
+```math
+\\begin{aligned}
+\\tilde{fd}_{yr,g,fd} &= \\left\\{{use}\\left(yr,i,j\\right) \\;\\vert\\; yr,\\, g \\in i,\\, fd \\in j \\right\\}
+\\\\
+\\tilde{fs}_{yr,g} &= \\left\\{\\tilde{fd}_{yr,g,fd} \\;\\vert\\; yr,\\, g \\in i,\\, fd = pce \\right\\}
+\\end{aligned}
+```
+
+```math
+\\begin{aligned}
+\\tilde{fs}_{yr,g} &= - \\min\\left\\{0, \\tilde{fs}_{yr,g} \\right\\}
+\\\\
+\\tilde{fd}_{yr,g,fd} &= \\max\\left\\{0, \\tilde{fd}_{yr,g,fd} \\right\\} \\;\\vert\\; yr,\\, g,\\, fd = pce \\right\\}
+\\end{aligned}
+```
+"""
+function _partition_fd!(d::Dict, set::Dict)
+    x = Rename(:s, :fd)
+    d[:fd0] = filter_with(edit_with(d[:use],x), set)
+    d[:fs0] = filter_with(d[:fd0], (fd=["pce","C"],); drop = true)
+    
+    d[:fs0][!,:value] .= - min.(0, d[:fs0][:,:value])
+
+    # For the sectoral disaggregation, 
+    if d[:sector]==:detail
```

Comment:
**For the future**. Backwards-compatible update for EEM: If we partition `fs0` and `fd0` in the same function, it's easier to add an option to say whether to drop negative values before/after calculating `a0`. I still want to pick one or the other to be consistent, but this is fine for now.